code/game/sound.dm

```diff
@@ -184,6 +184,31 @@ distance_multiplier - Can be used to multiply the distance at which the sound is
 			var/mob/M = m
 			M.playsound_local(M, null, volume, vary, frequency, null, channel, pressure_affected, S)
 
+/proc/play_soundtrack_music(sound_path, list/hearers = null, volume = 100, ignore_prefs = FALSE, play_to_lobby = FALSE, allow_deaf = TRUE, specific_z = null)
+	var/sound/S = sound(sound_path, volume=volume, wait=0, channel=CHANNEL_AMBIENT_MUSIC)
+	. = S
+
+	if(!hearers)
+		hearers = GLOB.player_list
+
+	for(var/mob/M in hearers)
+		if (!ismob(M))
+			continue
+
+		if (!ignore_prefs && !(M.client?.prefs?.toggles & SOUND_AMBIENCE))
+			continue
+
+		if (!play_to_lobby && isnewplayer(M))
+			continue
+
+		if (!allow_deaf && !M.can_hear())
+			continue
+
+		if (!isnull(specific_z) && M.z != specific_z)
```

Comment:
```suggestion
		if (M.z != specific_z)
```

Reply:
also this should check for virtual z too

Reply:
The is-null check is required, it shouldn't check for z at all if it's null

Reply:
you can compare null to value and it will be false